crates/wasmtime/src/component/func/host.rs

```diff
@@ -256,8 +279,8 @@ macro_rules! impl_into_component_func {
         impl<T, F, $($args,)* R> IntoComponentFunc<T, ($($args,)*), R> for F
         where
             F: Fn($($args),*) -> Result<R> + Send + Sync + 'static,
-            ($($args,)*): ComponentParams + Lift,
-            R: Lower,
+            ($($args,)*): ComponentParams + Lift + 'static,
+            R: Lower + 'static,
```

Comment:
Are these new `'static` bounds still required?

Reply:
It will be required now that we're making `HostFunc::typecheck` more than a no-op for the dynamic case, since I'll need to convert the type of that field back to a `Box<dyn Fn...>` so it can store a closure which captures the `TypeFuncIndex`.